test/Microsoft.Data.Sqlite.Tests/SqliteDataReaderTest.cs

```diff
@@ -293,6 +305,23 @@ public void GetFieldValue_of_DBNull_works()
                 DBNull.Value);
 
         [Fact]
+        public void GetFieldValue_of_DBNull_throws_when_not_null()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+
+                using (var reader = connection.ExecuteReader("SELECT 1;"))
+                {
+                    var hasData = reader.Read();
+
+                    Assert.True(hasData);
+                    Assert.Throws<InvalidCastException>(() => reader.GetFieldValue<DBNull>(0));
```

Comment:
In some extremely obscure case, one might be consider a breaking change.

Reply:
What's the case?

Reply:
`reader.GetFieldValue<DBNull>(0);` when the field is not `NULL`. Although I'd like to meet the person calling this with `DBNull`.